server/src/main/java/org/apache/calcite/avatica/server/AvaticaJsonHandler.java

```diff
@@ -106,14 +106,18 @@ public void handle(String target, Request baseRequest,
           // Avoid a new buffer creation for every HTTP request
           final UnsynchronizedBuffer buffer = threadLocalBuffer.get();
           try (ServletInputStream inputStream = request.getInputStream()) {
-            rawRequest = AvaticaUtils.readFully(inputStream, buffer);
+            byte[] bytes = AvaticaUtils.readFullyToBytes(inputStream, buffer);
+            String encoding = request.getCharacterEncoding();
```

Comment:
Shouldn't this part of getting the encoding and transforming the request be outside of the `if (rawRequest == null)`?

Reply:
What's the point of moving this outside of `if (rawRequest==null)`?

In case `rawRequest!=null`, we already have a DECODED `String`, thus we don't really need to play with encodings.

Reply:
The previous code was doing the transformation no matter what so I was just wondering if it was intended and I guess you answered my question.